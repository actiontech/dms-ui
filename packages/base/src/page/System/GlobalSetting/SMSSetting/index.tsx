import { useTranslation } from 'react-i18next';
import { useBoolean, useRequest } from 'ahooks';
import { useCallback, useMemo } from 'react';
import { Spin, Typography } from 'antd';
import { CustomLabelContent } from '@actiontech/shared/lib/components/CustomForm';
import ConfigExtraButtons from './components/ConfigExtraButtons';
import ConfigField from './components/ConfigField';
import { ConfigurationService } from '@actiontech/shared/lib/api';
import { IGetSmsConfigurationReplyItem } from '@actiontech/shared/lib/api/base/service/common';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { FormFields } from './index.type';
import { switchFieldName } from './index.data';
import {
  useConfigRender,
  ReadOnlyConfigColumnsType,
  ConfigSwitch,
  ConfigSubmitButtonField,
  useConfigSwitchControls
} from '@actiontech/shared';
import {
  PERMISSIONS,
  PermissionControl
} from '@actiontech/shared/lib/features';

const CodingSetting: React.FC = () => {
  const { t } = useTranslation();

  const [submitLoading, { setTrue: startSubmit, setFalse: submitFinish }] =
    useBoolean();

  const {
    form,
    renderConfigForm,
    startModify,
    modifyFinish,
    modifyFlag,
    extraButtonsVisible,
    enabled
  } = useConfigRender<FormFields>({
    switchFieldName,
    switchFieldLabel: (
      <CustomLabelContent
        title={t('dmsSystem.global.smsSetting.title')}
        tips=""
      />
    )
  });

  const {
    data: smsInfo,
    loading,
    refresh: refreshSmsInfo
  } = useRequest(
    () =>
      ConfigurationService.GetSmsConfiguration().then(
        (res) => res.data.data ?? {}
      ),
    {
      onSuccess(res) {
        if (res) {
          form.setFieldsValue({
            enabled: !!res.enable
          });
        }
      }
    }
  );

  const isConfigClosed = useMemo(() => {
    return !smsInfo?.enable;
  }, [smsInfo]);

  const setFormDefaultValue = useCallback(() => {
    form.setFieldsValue({
      url: smsInfo?.url,
      token: smsInfo?.configuration?.token
    });
  }, [form, smsInfo]);

  const handleClickModify = () => {
    setFormDefaultValue();
    startModify();
  };
  const handleClickCancel = () => {
    if (isConfigClosed) form.setFieldValue(switchFieldName, false);
    setFormDefaultValue();
    modifyFinish();
  };

  const onSubmit = (isCodingEnabled: boolean, values?: FormFields) => {
    startSubmit();
    ConfigurationService.UpdateSmsConfiguration({
      update_sms_configuration: {
        enable_sms: isCodingEnabled,
        url: values?.url,
        configuration: values?.token
          ? {
              token: values?.token ?? ''
            }
          : undefined
      }
    })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          modifyFinish();
          form.resetFields();
          refreshSmsInfo();
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  const onConfigSwitchPopoverConfirm = () => {
    if (!smsInfo?.enable && modifyFlag) {
      handleClickCancel();
      hiddenConfigSwitchPopover();
    } else {
      onSubmit(false);
    }
  };

  const {
    configSwitchPopoverOpenState,
    generateConfigSwitchPopoverTitle,
    onConfigSwitchPopoverOpen,
    handleConfigSwitchChange,
    hiddenConfigSwitchPopover
  } = useConfigSwitchControls(form, switchFieldName);

  const readonlyColumnsConfig: ReadOnlyConfigColumnsType<IGetSmsConfigurationReplyItem> =
    useMemo(() => {
      return [
        {
          label: t('dmsSystem.global.smsSetting.smsType'),
          span: 3,
          dataIndex: 'sms_type',
          hidden: !smsInfo?.enable,
          render: (val) => (
            <Typography.Paragraph>{val || '--'}</Typography.Paragraph>
          )
        },
        {
          label: 'url',
          span: 3,
          dataIndex: 'url',
          hidden: !smsInfo?.enable,
          render: (val) => (
            <Typography.Paragraph>{val || '--'}</Typography.Paragraph>
          )
        }
      ];
    }, [smsInfo, t]);

  return (
    <div className="config-form-wrapper">
      <Spin spinning={loading}>
        {renderConfigForm({
          data: smsInfo ?? {},
          columns: readonlyColumnsConfig,
          configExtraButtons: (
            <PermissionControl
              permission={
                PERMISSIONS.ACTIONS.BASE.SYSTEM.GLOBAL_SETTING.SMS_SERVICE
              }
            >
              <ConfigExtraButtons
                isConfigClosed={isConfigClosed}
                extraButtonsVisible={extraButtonsVisible}
                handleClickModify={handleClickModify}
                enabled={!!enabled}
              />
            </PermissionControl>
          ),
          configSwitchNode: (
            <PermissionControl
              permission={
                PERMISSIONS.ACTIONS.BASE.SYSTEM.GLOBAL_SETTING.SMS_SERVICE
              }
            >
              <ConfigSwitch
                title={generateConfigSwitchPopoverTitle(modifyFlag)}
                switchFieldName={switchFieldName}
                submitLoading={submitLoading}
                popoverVisible={configSwitchPopoverOpenState}
                onConfirm={onConfigSwitchPopoverConfirm}
                onSwitchChange={(open) => {
                  handleConfigSwitchChange(open, handleClickModify);
                }}
                onSwitchPopoverOpen={onConfigSwitchPopoverOpen}
              />
            </PermissionControl>
          ),
          configField: <ConfigField />,
          submitButtonField: (
            <ConfigSubmitButtonField
              submitLoading={submitLoading}
              handleClickCancel={handleClickCancel}
            />
          ),
          onSubmit: (values) => {
            onSubmit(enabled, values);
          }
        })}
      </Spin>
    </div>
  );
};

export default CodingSetting;
