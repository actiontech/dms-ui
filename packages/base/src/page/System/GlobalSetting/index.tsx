import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useBoolean, useRequest } from 'ahooks';
import useHideConfigInputNode from '../../../../../shared/lib/components/ConfigItem/hooks/useHideConfigInputNode';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { IUpdateSystemVariablesReqV1 } from '@actiontech/shared/lib/api/sqle/service/common';
import { ConfigFieldMapMeta } from '@actiontech/shared/lib/components/ConfigItem/index.type';
import { Spin } from 'antd';
import SystemBasicTitle from '../components/BasicTitle';
import OrderExpiredHours from './components/OrderExpiredHours';
import OperationRecordExpiredHours from './components/OperationRecordExpiredHours';
import UrlAddressPrefixTips from './components/UrlAddressPrefixTips';
import CBOperationLogsExpiredHours from './components/CBOperationLogsExpiredHours';
import { useCurrentUser } from '@actiontech/shared/lib/global';

const GlobalSetting = () => {
  const { t } = useTranslation();

  const { isAdmin } = useCurrentUser();

  const [
    orderExpiredHoursFieldVisible,
    {
      setTrue: showOrderExpiredHoursField,
      setFalse: hideOrderExpiredHoursField
    }
  ] = useBoolean();
  const [
    operationRecordExpiredHoursField,
    {
      setTrue: showOperationRecordExpiredHoursField,
      setFalse: hideOperationRecordExpiredHoursField
    }
  ] = useBoolean();

  const [
    cbOperationLogsExpiredHoursField,
    {
      setTrue: showCBOperationLogsExpiredHoursField,
      setFalse: hideCBOperationLogsExpiredHoursField
    }
  ] = useBoolean();

  const [urlFieldVisible, { setTrue: showUrlField, setFalse: hideUrlField }] =
    useBoolean();

  const hideFieldsAction = () => {
    if (orderExpiredHoursFieldVisible) hideOrderExpiredHoursField();
    if (operationRecordExpiredHoursField)
      hideOperationRecordExpiredHoursField();
    if (cbOperationLogsExpiredHoursField)
      hideCBOperationLogsExpiredHoursField();
    if (urlFieldVisible) hideUrlField();
  };

  const hasOneFieldVisible = useMemo(
    () =>
      orderExpiredHoursFieldVisible ||
      operationRecordExpiredHoursField ||
      urlFieldVisible,
    [
      orderExpiredHoursFieldVisible,
      operationRecordExpiredHoursField,
      urlFieldVisible
    ]
  );

  useHideConfigInputNode(hasOneFieldVisible, hideFieldsAction);

  const fieldMetaMap = new Map<
    keyof IUpdateSystemVariablesReqV1,
    ConfigFieldMapMeta
  >([
    [
      'workflow_expired_hours',
      {
        hideField: hideOrderExpiredHoursField
      }
    ],
    [
      'operation_record_expired_hours',
      {
        hideField: hideOperationRecordExpiredHoursField
      }
    ],
    [
      'cb_operation_logs_expired_hours',
      {
        hideField: hideCBOperationLogsExpiredHoursField
      }
    ],
    [
      'url',
      {
        hideField: hideUrlField
      }
    ]
  ]);

  const {
    data: globalConfig,
    loading: getConfigLoading,
    refresh
  } = useRequest(() =>
    configuration.getSystemVariablesV1().then((res) => res?.data?.data ?? {})
  );

  const [submitLoading, { setTrue: startSubmit, setFalse: submitFinish }] =
    useBoolean();

  const submitGlobalConfig = (
    value: string | number,
    fieldName: keyof IUpdateSystemVariablesReqV1
  ) => {
    const fieldMeta = fieldMetaMap.get(fieldName);

    startSubmit();
    configuration
      .updateSystemVariablesV1({
        ...globalConfig,
        [fieldName]: value
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          refresh();
          fieldMeta?.hideField?.();
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  return (
    <SystemBasicTitle title={t('dmsSystem.tabPaneTitle.globalConfiguration')}>
      <>
        <Spin spinning={getConfigLoading || submitLoading}>
          {/* main 分支中已删除该组件 */}
          <OrderExpiredHours
            expiredHours={globalConfig?.workflow_expired_hours}
            fieldVisible={orderExpiredHoursFieldVisible}
            showField={showOrderExpiredHoursField}
            hideField={hideOrderExpiredHoursField}
            submitGlobalConfig={submitGlobalConfig}
            isAdmin={isAdmin}
          />
          <OperationRecordExpiredHours
            expiredHours={globalConfig?.operation_record_expired_hours}
            fieldVisible={operationRecordExpiredHoursField}
            showField={showOperationRecordExpiredHoursField}
            hideField={hideOperationRecordExpiredHoursField}
            submitGlobalConfig={submitGlobalConfig}
          />
          <CBOperationLogsExpiredHours
            expiredHours={globalConfig?.cb_operation_logs_expired_hours}
            fieldVisible={cbOperationLogsExpiredHoursField}
            showField={showCBOperationLogsExpiredHoursField}
            hideField={hideCBOperationLogsExpiredHoursField}
            submitGlobalConfig={submitGlobalConfig}
          />
          <UrlAddressPrefixTips
            url={globalConfig?.url}
            fieldVisible={urlFieldVisible}
            showField={showUrlField}
            hideField={hideUrlField}
            submitGlobalConfig={submitGlobalConfig}
          />
        </Spin>
      </>
    </SystemBasicTitle>
  );
};

export default GlobalSetting;
