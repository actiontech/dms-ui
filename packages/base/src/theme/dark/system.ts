import { darkThemeUI } from '@actiontech/dms-kit/es/theme/dark/basic';
import { SystemTheme } from '../type';

export const systemTheme: SystemTheme = {
  configButton: {
    backgroundColor: darkThemeUI.uiToken.colorFillSecondary,
    border: `1px solid ${darkThemeUI.uiToken.colorBorderSecondary}`
  },
  logo: {
    boxShadow: '0 1px 4px 0 rgba(51, 44, 31, 0.12)'
  }
};
