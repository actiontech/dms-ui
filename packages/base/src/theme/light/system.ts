import { lightThemeUI } from '@actiontech/dms-kit/es/theme/light/basic';
import { SystemTheme } from '../type';

export const systemTheme: SystemTheme = {
  configButton: {
    backgroundColor: lightThemeUI.uiToken.colorFillSecondary,
    border: `1px solid ${lightThemeUI.uiToken.colorBorderSecondary}`
  },
  logo: {
    boxShadow: '0 1px 4px 0 rgba(51, 44, 31, 0.12)'
  }
};
