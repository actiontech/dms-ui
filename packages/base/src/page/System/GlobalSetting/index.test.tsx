import GlobalSetting from '.';

import system from '../../../testUtils/mockApi/system';

import { cleanup, fireEvent, act, screen } from '@testing-library/react';
import { superRender } from '@actiontech/shared/lib/testUtil/customRender';
import {
  getAllBySelector,
  getBySelector
} from '@actiontech/shared/lib/testUtil/customQuery';
import { createSpySuccessResponse } from '@actiontech/shared/lib/testUtil/mockApi';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';

describe('base/System/GlobalSetting', () => {
  let requestGetSystemVariables: jest.SpyInstance;
  let requestUpdateSystemVariables: jest.SpyInstance;

  const customRender = () => {
    return superRender(<GlobalSetting />);
  };

  beforeEach(() => {
    mockUseCurrentUser();
    jest.useFakeTimers();
    requestGetSystemVariables = system.getSystemVariables();
    requestUpdateSystemVariables = system.updateSystemVariables();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    cleanup();
  });

  it('render snap', async () => {
    const { baseElement } = customRender();
    await act(async () => jest.advanceTimersByTime(500));
    expect(baseElement).toMatchSnapshot();
    await act(async () => jest.advanceTimersByTime(2600));
    expect(requestGetSystemVariables).toHaveBeenCalled();
    expect(baseElement).toMatchSnapshot();
  });

  it('render hide form element node when click white place', async () => {
    const { baseElement } = customRender();

    await act(async () => jest.advanceTimersByTime(3200));

    const editBtn = getAllBySelector(
      '.config-item-filed-edit-button',
      baseElement
    );
    expect(editBtn.length).toBe(3);
    expect(editBtn[0]).toHaveAttribute('hidden');

    fireEvent.click(editBtn[1]);
    await act(async () => jest.advanceTimersByTime(500));
    fireEvent.mouseDown(screen.getByText('CB工作台操作审计过期时间(小时)'));
    await act(async () => jest.advanceTimersByTime(300));

    fireEvent.click(editBtn[2]);
    await act(async () => jest.advanceTimersByTime(500));
    fireEvent.mouseDown(screen.getByText('URL地址前缀'));
    await act(async () => jest.advanceTimersByTime(300));
  });

  it('render api return no data', async () => {
    requestGetSystemVariables.mockImplementation(() =>
      createSpySuccessResponse({})
    );
    const { baseElement } = customRender();
    await act(async () => jest.advanceTimersByTime(3100));
    expect(baseElement).toMatchSnapshot();
  });

  it('should hidden edit button when user is not admin', async () => {
    mockUseCurrentUser({ isAdmin: false });
    const { baseElement } = customRender();

    await act(async () => jest.advanceTimersByTime(3200));

    const editBtn = getAllBySelector(
      '.config-item-filed-edit-button',
      baseElement
    );
    expect(editBtn.length).toBe(3);
    expect(editBtn[0]).toHaveAttribute('hidden');
  });

  describe('render form validate error', () => {
    it('render operation_record_expired_hours', async () => {
      const { baseElement } = customRender();

      await act(async () => jest.advanceTimersByTime(3200));
      expect(screen.getByText('3161')).toBeInTheDocument();
      fireEvent.mouseOver(screen.getByText('3161'));
      await act(async () => jest.advanceTimersByTime(500));

      const editBtn = getAllBySelector(
        '.config-item-filed-edit-button',
        baseElement
      );
      fireEvent.click(editBtn[0]);
      await act(async () => jest.advanceTimersByTime(500));

      const inputNumEle = getBySelector('.ant-input-number-input', baseElement);
      fireEvent.change(inputNumEle, {
        target: {
          value: 'cef'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.change(inputNumEle, {
        target: {
          value: '9988'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.keyDown(inputNumEle, {
        key: 'Enter',
        keyCode: 13
      });
      await act(async () => jest.advanceTimersByTime(500));
      expect(requestUpdateSystemVariables).toHaveBeenCalled();
      await act(async () => jest.advanceTimersByTime(2600));
      expect(requestUpdateSystemVariables).toHaveBeenCalledWith({
        operation_record_expired_hours: 9988,
        url: 'http://demo.com',
        cb_operation_logs_expired_hours: 2160
      });
      await act(async () => jest.advanceTimersByTime(3000));
      expect(requestGetSystemVariables).toHaveBeenCalled();
    });

    it('render cb_operation_logs_expired_hours', async () => {
      const { baseElement } = customRender();

      await act(async () => jest.advanceTimersByTime(3200));
      expect(screen.getByText('2160')).toBeInTheDocument();
      fireEvent.mouseOver(screen.getByText('2160'));
      await act(async () => jest.advanceTimersByTime(500));

      const editBtn = getAllBySelector(
        '.config-item-filed-edit-button',
        baseElement
      );
      fireEvent.click(editBtn[1]);
      await act(async () => jest.advanceTimersByTime(500));

      const inputNumEle = getBySelector('.ant-input-number-input', baseElement);
      fireEvent.change(inputNumEle, {
        target: {
          value: 'cef'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.change(inputNumEle, {
        target: {
          value: '9988'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.keyDown(inputNumEle, {
        key: 'Enter',
        keyCode: 13
      });
      await act(async () => jest.advanceTimersByTime(500));
      expect(requestUpdateSystemVariables).toHaveBeenCalled();
      await act(async () => jest.advanceTimersByTime(2600));
      expect(requestUpdateSystemVariables).toHaveBeenCalledWith({
        operation_record_expired_hours: 3161,
        url: 'http://demo.com',
        cb_operation_logs_expired_hours: 9988
      });
      await act(async () => jest.advanceTimersByTime(3000));
      expect(requestGetSystemVariables).toHaveBeenCalled();
    });

    it('render url', async () => {
      const { baseElement } = customRender();

      await act(async () => jest.advanceTimersByTime(3200));
      expect(screen.getByText('http://demo.com')).toBeInTheDocument();
      fireEvent.mouseOver(screen.getByText('http://demo.com'));
      await act(async () => jest.advanceTimersByTime(500));

      const editBtn = getAllBySelector(
        '.config-item-filed-edit-button',
        baseElement
      );
      fireEvent.click(editBtn[2]);
      await act(async () => jest.advanceTimersByTime(500));

      const inputEle = getBySelector('#editInput', baseElement);
      fireEvent.change(inputEle, {
        target: {
          value: 'this is a url string'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.keyDown(inputEle, {
        key: 'Enter',
        keyCode: 13
      });
      await act(async () => jest.advanceTimersByTime(500));
      expect(requestUpdateSystemVariables).toHaveBeenCalled();
      await act(async () => jest.advanceTimersByTime(2600));
      expect(requestUpdateSystemVariables).toHaveBeenCalledWith({
        operation_record_expired_hours: 3161,
        url: 'this is a url string',
        cb_operation_logs_expired_hours: 2160
      });
      await act(async () => jest.advanceTimersByTime(3000));
      expect(requestGetSystemVariables).toHaveBeenCalled();
    });
  });
});
