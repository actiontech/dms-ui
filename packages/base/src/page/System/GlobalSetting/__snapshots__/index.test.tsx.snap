// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/System/GlobalSetting render api return no data 1`] = `
<body>
  <div>
    <section
      class="system-form-wrapper"
    >
      <div
        class="ant-row ant-row-space-between ant-row-middle config-title-wrapper has-border css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
          >
            全局配置
          </div>
        </div>
      </div>
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  操作记录过期时间(小时)
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                -
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  URL地址前缀
                </div>
                <div
                  class="tips"
                >
                  配置能访问SQLE的url地址信息, 格式为 http(s)://ip:port/sqle 
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                -
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`base/System/GlobalSetting render snap 1`] = `
<body>
  <div>
    <section
      class="system-form-wrapper"
    >
      <div
        class="ant-row ant-row-space-between ant-row-middle config-title-wrapper has-border css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
          >
            全局配置
          </div>
        </div>
      </div>
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
      >
        <div>
          <div
            aria-busy="true"
            aria-live="polite"
            class="ant-spin ant-spin-spinning css-dev-only-do-not-override-txh9fw"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  操作记录过期时间(小时)
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                -
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  URL地址前缀
                </div>
                <div
                  class="tips"
                >
                  配置能访问SQLE的url地址信息, 格式为 http(s)://ip:port/sqle 
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                -
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`base/System/GlobalSetting render snap 2`] = `
<body>
  <div>
    <section
      class="system-form-wrapper"
    >
      <div
        class="ant-row ant-row-space-between ant-row-middle config-title-wrapper has-border css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
          >
            全局配置
          </div>
        </div>
      </div>
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  操作记录过期时间(小时)
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                3161
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-space-between ant-row-middle css-19fu9ar css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-col ant-col-11 config-item-label css-dev-only-do-not-override-txh9fw"
            >
              <div
                class="config-item-label"
              >
                <div
                  class="title"
                >
                  URL地址前缀
                </div>
                <div
                  class="tips"
                >
                  配置能访问SQLE的url地址信息, 格式为 http(s)://ip:port/sqle 
                </div>
              </div>
            </div>
            <div
              class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center config-item-filed"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                http://demo.com
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-icon-only config-item-filed-edit-button basic-button-wrapper css-geipcv"
                  hidden=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <svg
                      fill="none"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="edit-2-fill"
                      >
                        <path
                          d="M6.162 12.6667H14V14H2V11.1713L8.6 4.57133L11.428 7.40066L6.16133 12.6667H6.162ZM9.542 3.62933L10.9567 2.21466C11.0817 2.08968 11.2512 2.01947 11.428 2.01947C11.6048 2.01947 11.7743 2.08968 11.8993 2.21466L13.7853 4.10066C13.9103 4.22568 13.9805 4.39522 13.9805 4.57199C13.9805 4.74877 13.9103 4.91831 13.7853 5.04333L12.3707 6.45733L9.54267 3.62933H9.542Z"
                          fill="#8A8F99"
                          id="Vector"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
